virl2_client/models/auth_management.py

```diff
@@ -0,0 +1,392 @@
+        # might as well save a request and store said settings in the relevant manager
+        if (manager := self._managers[self._method]) is not None:
+            manager._settings = settings
+            manager._last_sync_time = time.time()
+
+    @property
+    def method(self) -> AuthMethod:
+        self.sync_if_outdated()
+        return self._method
+
+    @method.setter
+    def method(self, value: AuthMethod | str):
+        value = AuthMethod(value)
+        self._session.put(self._METHOD_URL, json={"method": value.value})
+        self._method = value
+
+    @property
+    def manager(self) -> AuthMethodManager:
+        """
+        The manager for the current auth method.
+        """
+        # TODO: should we allow users to also configure currently inactive auth methods?
+        #  not sure how useful that might or might not be, but if they can then that
+        #  changes the answer to the question in AuthMethodManager.update_setting
+        self.sync_if_outdated()
+        return self._managers[self._method]
+
+
+class AuthMethodManager:
+    _METHOD = ""
+    _CONFIG_URL = _BASE_URL + "/config"
+    _TEST_URL = _BASE_URL + "/test"
+
+    def __init__(self, auth_management: AuthManagement):
+        self._auth_management = auth_management
+        self._last_sync_time = 0.0
+        self._settings = {}
+
+    def sync_if_outdated(self) -> None:
+        timestamp = time.time()
+        if (
+            self._auth_management.auto_sync
+            and timestamp - self._last_sync_time
+            > self._auth_management.auto_sync_interval
+        ):
+            self.sync()
+
+    def sync(self) -> None:
+        self._settings = self._get_current_settings()
+        self._last_sync_time = time.time()
+
+    @property
+    def _session(self) -> Client:
+        return self._auth_management._session
+
+    def _get_current_settings(self) -> dict:
+        settings = self._session.get(self._CONFIG_URL).json()
+        del settings["method"]
+        return settings
+
+    def get_setting(self, setting: str) -> Any:
+        """
+        Returns the value of a specific setting of this authentication method.
+        Note: consider using parameters instead.
+        """
+        self.sync_if_outdated()
+        return self._settings[setting]
+
+    def get_settings(self) -> dict:
+        """Returns a dictionary of the settings of this authentication method."""
+        self.sync_if_outdated()
+        return self._settings.copy()
+
+    def update_setting(self, setting: str, value) -> None:
+        """
+        Update a setting for this auth method.
+
+        :param setting: the setting to update
+        :param value: the value to set the setting to
+        """
+        if setting == "method":
+            raise ValueError(
+                f"To set method, use 'AuthManagement.method = {value}' instead."
+            )
+        # TODO: currently, method is required, so we have to provide it;
```

Comment:
We should think 2 steps ahead in this case. Since there is a discussion about updating API schema and create separate schema for each method, which means each auth method is going to have its own (required) properties, it doesn't make any sense to allow user to update method WITHOUT updating every other property, if that's what you are referring to.

If you are talking about completely switching auth_manager when you switch method, I guess that makes sense, but as we already discussed, I would like to switch update_settings to private method and only allow user to change method by method property, which directly fixes this issue of yours.